tests/conftest.py

```diff
@@ -917,6 +920,52 @@ def create_mock_mqtt(*args, **kwargs) -> MqttMockHAClient:
         yield _setup_mqtt_entry
 
 
+@pytest.fixture
+def yaml_config() -> ConfigType | None:
+    """Fixture to override configuration.yaml using mock_yaml_config.
+
+    Patches the content of configuration.yaml with the
+    yaml dump of a config dict.
+    """
+    return None
+
+
+@pytest.fixture
+def yaml_configuration(yaml_config: ConfigType | None) -> str | None:
```

Comment:
Why is one fixture name `yaml_config` and another `yaml_configuration`? Can they get better names to distinguish them?
Also, same comment about the misleading docstring.

Reply:
I wonder if we should have two distinct mocking paths:
path 1 => would set YAML strings and run the YAML validation
- `yaml_configuration` fixture returns a `str`
- `yaml_configuration_files` fixture returns a `dict[str, str]`
- `mock_yaml_configuration` fixture uses `patch_yaml_files`

path 2 => would set YAML ConfigType and skip the YAML validation
- `hass_config` fixture returns a `ConfigType`
- `mock_hass_config` fixture patches `homeassistant.config.async_hass_config_yaml`

Reply:
Two paths maybe makes sense, but the naming and docstrings should be clear and concise regardless of if it's one or two paths.

Reply:
- `yaml_configuration` was renamed to `hass_config_yaml`
- `yaml_configuration_files` was renamed to `hass_config_yaml_files`
- `mock_yaml_configuration` was renamed to `mock_hass_config_yaml`